frappe/database/query.py

```diff
@@ -346,7 +347,42 @@ def dict_query(self, table: str, filters: dict[str, str | int] = None, **kwargs)
 			if not isinstance(key, str):
 				conditions = conditions.where(self.make_function_for_filters(key, value))
 				continue
+			# Nested set support
 			if isinstance(value, (list, tuple)):
+				if value in self.OPERATOR_MAP["nested_set"]:
```

Comment:
separate the nested set logic into a separate function? this seems too stuffed up